fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCParameterCacheAdapter.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * (C) Copyright IBM Corp. 2022
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+ 
+package com.ibm.fhir.persistence.jdbc.util;
+
+import com.ibm.fhir.persistence.jdbc.FHIRPersistenceJDBCCache;
+import com.ibm.fhir.persistence.jdbc.dao.api.LogicalResourceIdentKey;
+import com.ibm.fhir.persistence.jdbc.dto.CommonTokenValue;
+import com.ibm.fhir.persistence.params.api.IParameterIdentityCache;
+
+/**
+ * An adapter to implement the {@link IParameterIdentityCache} using the {@link FHIRPersistenceJDBCCache}
+ */
+public class JDBCParameterCacheAdapter implements IParameterIdentityCache {
+
+    // The delegate cache
+    private final FHIRPersistenceJDBCCache cache;
+
+    /**
+     * Public constructor
+     * @param cache
+     */
+    public JDBCParameterCacheAdapter(FHIRPersistenceJDBCCache cache) {
+        this.cache = cache;
+    }
+
+    @Override
+    public Integer getParameterNameId(String parameterName) {
+        return cache.getParameterNameCache().getId(parameterName);
+    }
+
+    @Override
+    public Integer getCodeSystemId(String codeSystem) {
+        return cache.getResourceReferenceCache().getCodeSystemId(codeSystem);
+    }
+
+    @Override
+    public Long getCommonTokenValueId(short shardKey, String codeSystem, String tokenValue) {
+        return cache.getResourceReferenceCache().getCommonTokenValueId(codeSystem, tokenValue);
+    }
+
+    @Override
+    public void addParameterName(String parameterName, int parameterNameId) {
+        // TODO
```

Comment:
does this need to get done?

Reply:
ooh, good catch. Yes I missed this (although it's only performance, because the value will get added to the cache the next time it is requested)